### PR TITLE
DIGEST-MD5 auth

### PR DESCRIPTION
**Pull request by uzbugg**
For users who are unable to authenticate against SMTP.

### CODE REVIEW

**Comment by euantorano** [comment]
Closing due to #2799 which was merged earlier.

**Comment by euantorano** [comment]
#2799 solved the CRAM-MD5 but not Digest MD5. This PR currently has conflicts though.

File: inc/mailhandlers/digestmd5.php [hunk 1]
```diff
@@ -0,0 +1,179 @@
+<?php
+// +-----------------------------------------------------------------------+ 
+// | Copyright (c) 2002-2003 Richard Heyes                                 |
```
Comment (effone) [inline inc/mailhandlers/digestmd5.php]: All merged codes should be copyrighted to MyBB only being you are a contributor to it ...

File: inc/mailhandlers/digestmd5.php [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+// $Id$
+/**
+* Implmentation of DIGEST-MD5 SASL mechanism
+*
+* @author  Richard Heyes <richard@php.net>
+* @access  public
+* @version 1.0
+* @package Auth_SASL
+*/
+//require_once('Common.php');
```
Comment (effone) [inline inc/mailhandlers/digestmd5.php]: Where is the common.php?

File: inc/mailhandlers/digestmd5.php [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    * @param  string $service   The servicename (eg. imap, pop, acap etc)
+    * @param  string $authzid   Authorization id (username to proxy as)
+    * @return string            The digest response (NOT base64 encoded)
+    * @access public
+    */
+    function getResponse($authcid, $pass, $challenge, $hostname, $service, $authzid = '')
+    {
+        $challenge = $this->_parseChallenge($challenge);
+        $authzid_string = '';
+        if ($authzid != '') {
```
Comment (effone) [inline inc/mailhandlers/digestmd5.php]: Please follow MyBB coding standard

File: inc/mailhandlers/digestmd5.php [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+            $authzid_string = ',authzid="' . $authzid . '"'; 
+        }
+        if (!empty($challenge)) {
```
Comment (effone) [inline inc/mailhandlers/digestmd5.php]: Please follow MyBB coding standard

File: inc/mailhandlers/digestmd5.php [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+            $cnonce         = $this->_getCnonce();
+            $digest_uri     = sprintf('%s/%s', $service, $hostname);
+            $response_value = $this->_getResponseValue($authcid, $pass, $challenge['realm'], $challenge['nonce'], $cnonce, $digest_uri, $authzid);
+            if ($challenge['realm']) {
```
Comment (effone) [inline inc/mailhandlers/digestmd5.php]: Please follow MyBB coding standard

File: inc/mailhandlers/digestmd5.php [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+                return sprintf('username="%s",realm="%s"' . $authzid_string  .
+',nonce="%s",cnonce="%s",nc=00000001,qop=auth,digest-uri="%s",response=%s,maxbuf=%d', $authcid, $challenge['realm'], $challenge['nonce'], $cnonce, $digest_uri, $response_value, $challenge['maxbuf']);
+            } else {
```
Comment (effone) [inline inc/mailhandlers/digestmd5.php]: Please follow MyBB coding standard

**Pull request closed without merging**